src/common/Kokkos_ArithTraits.hpp

```diff
@@ -225,6 +225,230 @@ KOKKOS_FORCEINLINE_FUNCTION IntType intPowUnsigned(const IntType x,
+  static KOKKOS_FUNCTION mag_type abs(const val_type x) { return x; }
+
+#define KOKKOSKERNELS_ARITHTRAITS_INTEGRAL(KOKKOSKERNELS_ABS)                 \
+  static KOKKOS_FUNCTION val_type zero() { return static_cast<val_type>(0); } \
+  static KOKKOS_FUNCTION val_type one() { return static_cast<val_type>(1); }  \
+  static KOKKOS_FUNCTION val_type min() {                                     \
+    return Kokkos::Experimental::finite_min<val_type>::value;                 \
+  }                                                                           \
+  static KOKKOS_FUNCTION val_type max() {                                     \
+    return Kokkos::Experimental::finite_max<val_type>::value;                 \
+  }                                                                           \
+  static KOKKOS_FUNCTION val_type infinity() {                                \
+    return static_cast<val_type>(0);                                          \
+  }                                                                           \
+  static KOKKOS_FUNCTION bool isInf(const val_type) { return false; }         \
+  static KOKKOS_FUNCTION bool isNan(const val_type) { return false; }         \
+  KOKKOSKERNELS_ABS                                                           \
+  static KOKKOS_FUNCTION mag_type real(const val_type x) { return x; }        \
+  static KOKKOS_FUNCTION mag_type imag(const val_type) { return zero(); }     \
+  static KOKKOS_FUNCTION val_type conj(const val_type x) { return x; }        \
+  static KOKKOS_FUNCTION val_type pow(const val_type x, const val_type y) {   \
+    return Kokkos::pow(x, y);                                                 \
+  }                                                                           \
+  static KOKKOS_FUNCTION val_type sqrt(const val_type x) {                    \
+    return static_cast<val_type>(Kokkos::sqrt(abs(x)));                       \
+  }                                                                           \
+  static KOKKOS_FUNCTION val_type cbrt(const val_type x) {                    \
+    return static_cast<val_type>(Kokkos::cbrt(abs(x)));                       \
+  }                                                                           \
+  static KOKKOS_FUNCTION val_type exp(const val_type x) {                     \
+    return static_cast<val_type>(Kokkos::exp(abs(x)));                        \
+  }                                                                           \
+  static KOKKOS_FUNCTION val_type log(const val_type x) {                     \
+    return static_cast<val_type>(Kokkos::log(abs(x)));                        \
+  }                                                                           \
+  static KOKKOS_FUNCTION val_type log10(const val_type x) {                   \
+    return static_cast<val_type>(Kokkos::log10(abs(x)));                      \
+  }                                                                           \
```

Comment:
Why are we taking the absolute value?!!  (I did see that this is how it currently is implemented)